Source/WebInspectorUI/UserInterface/Views/ScreenshotsTimelineOverviewGraph.js

```diff
@@ -57,24 +66,34 @@ WI.ScreenshotsTimelineOverviewGraph = class ScreenshotsTimelineOverviewGraph ext
 
         let secondsPerPixel = this.timelineOverview.secondsPerPixel;
 
+        if (this._lastSelectedRecordInLayout)
+            this._imageElementForRecord.get(this._lastSelectedRecordInLayout)?.classList.remove("selected");
+
         this._lastSelectedRecordInLayout = this.selectedRecord;
 
+        if (this._lastSelectedRecordInLayout)
+            this._imageElementForRecord.get(this._lastSelectedRecordInLayout)?.classList.add("selected");
+
```

Comment:
> Should this be below the `getOrInitialize` loop below so that we add the style even if the image element didn't exist when we started layout? I think otherwise we also need to check each record to see if its the selected record when we create the `img` element.

oops yes great catch will adjust

> For example, if you take a recording, select a record, go to a different tab and GC cleans up the image elements, and then return to the tab the selection should remain, but the element itself may no longer exist in the map.

The only way for GC to collect the `<img>` is if the related `WI.ScreenshotsTimelineRecord` is also destroyed (i.e. the timeline is cleared).  `WeakMap` keeps the value alive so long as the key is alive.

Reply:
🤦 Sorry, for some reason I always get which part of the map is weakly held backwards in my head. I do the same thing with WTF::WeakHashMap as well.